docs/SPIR-V.rst

```diff
@@ -0,0 +1,290 @@
+
+=========================== =================
+HLSL Interpolation Modifier SPIR-V Decoration
+=========================== =================
+``linear``                  <none>
+``centroid``                ``Centroid``
+``nointerpolation``         ``Flat``
+``noperspective``           ``NoPerspective``
+``sample``                  ``Sample``
+=========================== =================
+
+User-defined types
+++++++++++++++++++
+
+`User-defined types <https://msdn.microsoft.com/en-us/library/windows/desktop/bb509702(v=vs.85).aspx>`_ are type aliases introduced by typedef. No new types are introduced and we can rely on Clang to resolve to the original types.
+
+Samplers and textures
++++++++++++++++++++++
+
+[TODO]
+
+Buffers
++++++++
+
+[TODO]
+
+Variables and resources
+-----------------------
+
+Definition
+++++++++++
+
+Variables are defined in HLSL using the following `syntax <https://msdn.microsoft.com/en-us/library/windows/desktop/bb509706(v=vs.85).aspx>`_ rules::
+
+  [StorageClass] [TypeModifier] Type Name[Index]
+      [: Semantic]
+      [: Packoffset]
+      [: Register];
+      [Annotations]
+      [= InitialValue]
+
+Storage class
++++++++++++++
+
+[TODO]
+
+Type modifier
++++++++++++++
+
+[TODO]
+
+Interface variables
++++++++++++++++++++
+
+Direct3D uses "`semantics <https://msdn.microsoft.com/en-us/library/windows/desktop/bb509647(v=vs.85).aspx>`_" to compose and match the interfaces between subsequent stages. These semantics modifiers can appear after struct members, global variables, and also function parameters and return values. E.g.,::
+
+  struct VSInput {
+    float4 pos  : POSITION;
+    float3 norm : NORMAL;
+    float4 tex  : TEXCOORD0;
+  };
+
+  float4 pos: SV_POSITION;
+
+  float4 VSFunction(float4 pos : POSITION) : POSITION {
+    return pos;
+  }
+
+In Clang AST, these semantics are represented as ``SemanticDecl``, which is attached to the corresponding struct members (``FieldDecl``), global variables (``VarDecl``), and function parameters (``ParmVarDecl``) and return values (``FunctionDecl``).
+
+[TODO] How to map semantics to SPIR-V interface variables
+
+Expressions
+-----------
+
+[TODO]
+
+Control flows
+-------------
+
+[TODO]
+
+Builtin functions
+-----------------
+
+[TODO]
+
+Logistics
+=========
+
+Project planning
+----------------
+
+We use `GitHub Project feature in the Google fork repo <https://github.com/google/DirectXShaderCompiler/projects/1>`_ to manage tasks and track progress.
+
+Pull requests and code review
+-----------------------------
+
+Pull requests are very welcome! However, the Google repo is only used for project planning. We do not intend to maintain a detached fork; so all pull requests should be sent against the original `Microsoft repo <https://github.com/Microsoft/DirectXShaderCompiler>`_. Code reviews will also happen there.
+
+For each pull request, please make sure
+
+- Indicating your intension in the Google fork to avoid duplicate work.
```

Comment:
To make this fit grammatically with the "please make sure", I suggest:

"Indicating your intension in the"  ->  "You express your intent in the"

How do they express their intent?   Are you going to use a similar process as described in https://github.com/KhronosGroup/SPIRV-Tools/blob/master/projects.md ?   You can leave this for now, but can do a followup MR to capture this detail.